### DISCUSSION TITLE
Missing output shape when using infer_shapes_path on a model with external data?

### DISCUSSION THREAD

**Discussion by daveliddell** [onnx/onnx#5876] [Q&A]
Hi!  I'm having trouble converting code that previously used `infer_shapes ` to use `infer_shapes_path` instead because the model I'm working with is too large for `infer_shapes`.  However, on a small test model, it seems that now the output shape isn't getting set.  I'm using onnx 1.15 with python 3.10 on ubuntu 22 or 20 (doesn't seem to matter).

Here's the model (with data externalized):

```
def create_model(model_file: Path, data_path: Path):
    const = make_node('Constant', [], ['c_shape'], 'const',
        value=numpy_helper.from_array(numpy.array([4], dtype=numpy.int64)))
    cofshape = make_node('ConstantOfShape', ['c_shape'], ['c_out'], 'cofshape',
        value=numpy_helper.from_array(numpy.array([1], dtype=numpy.int64)))

    outval = make_tensor_value_info('c_out', TensorProto.INT64, [None])
    graph = make_graph([const, cofshape], 'constgraph', [], [outval])

    onnx_model = make_model(graph)
    convert_model_to_external_data(onnx_model, all_tensors_to_one_file=True,
        location=data_path, size_threshold=0, convert_attribute=True)
    onnx.save(onnx_model, model_file)

```

Here's the code to load it and infer shapes:
```
    onnx.shape_inference.infer_shapes_path(args.input_file, temp_file)
    inferred_model = onnx.load(temp_file, load_external_data=False)
    onnx.load_external_data_for_model(inferred_model, data_dir)
```

Here's a dump of the output of the resulting model:
```
  output {
    name: "c_out"
    type {
      tensor_type {
        elem_type: 7
        shape {
          dim {   <-- was expecting "dim_value: 4" here
          }
        }
      }
    }
  }
 ```

Some experiments I've tried:

- infer_shapes_path + internal data --> PASS
- infer_shapes_path + external data --> FAIL
- infer_shapes + external data --> PASS

Here's how we were using infer_shapes before:
```
    raw_model = onnx.load(file_path)
    inferred_model = onnx.shape_inference.infer_shapes(raw_model)
```

Before I start debugging, I was hoping someone might at least point me in the right direction.  Thanks!

**Comment by leshabirukov** [comment]
I believe, the problem is, in `make_graph` you place `outval` to `output` parameter instead of `value_info`. Try this change:

    graph = make_graph([const, cofshape], 'constgraph', [],[], value_info=[outval])
